iceoryx_posh/source/mepoo/mem_pool.cpp

```diff
@@ -101,14 +101,14 @@ void* MemPool::getChunk() noexcept
     return indexToPointer(index, m_chunkSize, m_rawMemory.get());
 }
 
-void* MemPool::indexToPointer(uint32_t index, uint32_t chunkSize, void* const rawMemoryBase) noexcept
+void* MemPool::indexToPointer(uint32_t index, uint64_t chunkSize, void* const rawMemoryBase) noexcept
 {
     const auto offset = static_cast<uint64_t>(index) * chunkSize;
     return static_cast<void*>(static_cast<uint8_t*>(rawMemoryBase) + offset);
```

Comment:
This is more of a comment on the previous change to fix overflow for memory pools spanning more than 4GB. 

Do we have to care about 32-bit operating systems? I.e. should we be using `size_t` and validating while parsing the config and via a few `IOX_ENSURES` statements that we don't overflow `numeric_limits<size_t>::max`? 

Otherwise the pointer arithmetic will simply overflow on those platforms and probably segfault.

Reply:
In any case I wouldn't require any action in this particular PR. It should be a separate issue.

Reply:
These methods could also easily become private. The only reason they are public is because it was easier to test.